feat(schema): add non-PHI healthcare VC examples and validation

Where-Mental-Wellness-Meets-Metaverse opened the pull request:
This PR adds two minimal, non-PHI healthcare Verifiable Credential examples to the documentation and examples section, including:

- `MentalWellnessSessionCredential` (attendance proof without PHI)
- `PractitionerAttestationCredential` (practitioner status attestation with license hash and revocation support)

Also includes:
- Privacy guidance on PHI minimization, selective disclosure, and key rotation
- Sample credentials for each schema
- Automated JSON Schema validation using AJV

Purpose: Provide healthcare-focused VC examples that are privacy-safe and compliance-friendly, which can be reused or adapted in related projects.

Where-Mental-Wellness-Meets-Metaverse commented:
Thanks for the review. I’ve applied the requested updates:
   
   -    
Removed “All rights reserved” and aligned with Apache-2.0 across docs, samples, and tests.

   -    
Removed node_modules and added it to .gitignore.

   -    
Added npm install / npm run test instructions to the README for the tests folder.

   -    
Fixed tests config by removing "type": "module" from package.json.

   -    
Renamed the practitioner sample to include .json.

   -    
Removed the stray .save file and corrected a typo.

These are non-PHI healthcare example schemas meant to showcase Aries VCR’s versatility in privacy-preserving credential design. By modeling healthcare use cases that avoid PHI, they demonstrate how Aries can support regulated-domain credentials with selective disclosure and compliance-friendly patterns — an approach equally applicable to finance, education, or other sensitive sectors.


---
    On Thursday, August 14, 2025 at 05:31:52 PM EDT, Emiliano Suñé ***@***.***> wrote:  
 
 


@esune requested changes on this pull request.

Thank you for the contribution @Where-Mental-Wellness-Meets-Metaverse.
I added a couple of comments about licensing - I am no legal expert but think that the wording should be removed or changed to reflect the Apache2 license used for the repository.

It also looks like you committed the node_modules folder to the repository: that should be removed and added to .gitignore to avoid pushing it to Git, and instead instructions should be added to the README file (likely just running npm install and npm run test from within the tests folder, I think).

Finally, I do wonder if this repository is the right place to save this documentation/examples since: can you provide any information about how you are planning to link to these resources? Mostly curious to understand how these W3C credentials would/could be used: based on the name they seem like they could be "persona;" credentials in a personal wallet, rather than a "community repository" like Aries VCR.

In docs/healthcare-vc-examples/README.md:
> @@ -0,0 +1,14 @@
+# Healthcare Verifiable Credential Examples (Non-PHI)
+
+All rights reserved, unlicensed.

I am not sure this matches the requirements of the repository, which uses the Apache 2 license. If the content was copied/moved from elsewhere, then we may not be able to accept it. If not, this should be removed.

In docs/healthcare-vc-examples/privacy_healthcare_credentials.md:
> @@ -0,0 +1,20 @@
+# Privacy & Compliance Notes for Health Credentials (Non-PHI)
+
+All rights reserved, unlicensed.

Same as previous comment.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>

esune commented:
@Where-Mental-Wellness-Meets-Metaverse the `node_modules` folder is still committed to the repo, the files need to be removed from the git index and the change committed.

DCO should also be fixed as it is required by the repo.

Where-Mental-Wellness-Meets-Metaverse commented:
Addressed feedback: removed committed node_modules, added global ignore, ensured DCO on all commits, and clarified repo fit/usage in docs/healthcare-vc-examples/README.md. Please re-review.

Where-Mental-Wellness-Meets-Metaverse commented:
I applied changes:

Replaced:
   
   -    
client/node_modules

   -    
node_modules/

   -    
**/node_modules/ (duplicate)


With:
   
   -    
client/

   -    
**/node_modules/


No other changes. Commit: a9e42d1 on feat/healthcare-vcs-nonphi.


   


 On Monday, August 18, 2025 at 11:47:00 AM EDT, Wade Barnes ***@***.***> wrote:  
 
 


@WadeBarnes requested changes on this pull request.

Please update the .gitignore file to restore the existing functionality and remove the redundant node_module entries.

In .gitignore:
> +client/node_modules
+node_modules/
+**/node_modules/
+
+**/node_modules/
⬇️ Suggested change-client/node_modules
-node_modules/
-**/node_modules/
-
-**/node_modules/
+client/
+
+**/node_modules/

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>

WadeBarnes reviewed (changes requested):
Thanks for fixing that.  Please update the last commit with DCO.  I think that should be the final change.

esune commented on README.md:
````diff
@@ -81,3 +81,11 @@ We also welcome issues submitted about problems you encounter in using Aries VCR
 ## License
 
 [Apache License Version 2.0](https://github.com/bcgov/aries-vcr/blob/main/LICENSE)
+
+### Testing
+
+```bash
+npm install
+npm run test
+```
````
This block should go in `docs/healthcare-vc-examples/README.md` instead, since it is specific to that set of documents and not the aries-vcr in general.

esune reviewed (changes requested):
Caught a last inconsistency in the readme instructions (wrong readme file), I think we're good to go after that is fixed.

Where-Mental-Wellness-Meets-Metaverse commented:
DCO has been added to the last commit and the branch has been rebased on upstream/main. Ready for review. 






    On Monday, August 25, 2025 at 09:12:43 AM EDT, Wade Barnes ***@***.***> wrote:  
 
 


@WadeBarnes requested changes on this pull request.

@Where-Mental-Wellness-Meets-Metaverse, Your last commit is missing DCO, please amend. Thanks.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>
